### DISCUSSION TITLE
Reaktywacja

### DISCUSSION THREAD

**Discussion by olsza** [davpal/progit2-pl#24] [General]
Jak to wspomninał @davpal w https://github.com/progit2-pl/progit2-pl/issues/19#issue-1625482476, nasze repo było bardzo nie aktualne z oryginalnym źródłem, m.in. zmienił się częściowo schemat katalogów. Postanowiłem utworzyć kopię branch-a `main` do którego będziemy "wrzucać" tłumaczenia.

Aktualne tłumaczenia z `master` zostaną przeniesione do `main` z uwzględnieniem poprawek, jakie nastąpiły w oryginale EN.

Do momentu pełnego lub prawie pełnego przeniesienia tłumaczeń i funkcjonalności na `main`, domyślnym i jednocześnie aktualnym branch-em do zgłoszeń PR będzie `master`.

Wszelka pomoc do przeniesienia z `master` do `main` jest mile widziana :)

**Comment by olsza** [comment]
Zależy mi aby zachować commit-y przy przenoszeniu tłumaczeń do 'main' z aktualnego branch-u 'master', gdyż później będzie można automatycznie wyciągać autorów i dołączyć do książki osoby, które pomagały przy wydaniu polskiej edycji. 

Jak uważacie za najlepsze sposób łączenia i zachowaniem historii na pojedynczych plikach?

Bo próbuje kilka i w sumie w każdym mam ten sam problem, że jest masowe łączenie i przy tym na dziesiątkach plikach jest konflikt, które musze sam rozwiązywać, a chciałbym aby każdy kto chce tu pomoc mógł np. wybierać plik rozwiązać konflikt i poprawić ewentualne drobne poprawki, które poprawiono w oryginale a w edycji PL już nie.

**Reply by davpal** [reply]
Zastanawiałem się nad tym i jedyne co mi przyszło do głowy:

Wybieramy sobie jakiś commit gdzie było tłumaczenie na `origin/master` i uzywamy cherry-pick lecz z opcja `--no-commit`
`git cherry-pick --no-commit 1d32bd84467f316e628dbb186dca681b3250a0a3`

Na wybranym pliku mozna rozwiazać konflikty, cos poprawić a reszte plików na które na ten moment nie mamy ochoty potraktować przez reset & checkout:
```
$ git status

On branch main

Your branch is up to date with 'progit/main'.

Unmerged paths:
  (use "git restore --staged <file>..." to unstage)
  (use "git add/rm <file>..." as appropriate to mark resolution)

	both modified:   book/05-distributed-git/sections/contributing.asc
	both modified:   book/07-git-tools/sections/debugging.asc
	both modified:   book/08-customizing-git/sections/config.asc
	deleted by us:   book/09-git-and-other-scms/1-git-and-other-scms.asc
	both modified:   book/10-git-internals/sections/maintenance.asc

#
# Rozwiazałem o drodze konflikt do contributing.asc
# 
$ git add book/05-distributed-git/sections/contributing.asc

$ git reset book/07-git-tools/sections/debugging.asc  book/08-customizing-git/sections/config.asc  book/09-git-and-other-scms/1-git-and-other-scms.asc  book/10-git-internals/sections/maintenance.asc
$ git checkout book/07-git-tools/sections/debugging.asc  book/08-customizing-git/sections/config.asc   book/10-git-internals/sections/maintenance.asc 

$ git status
On branch main
Your branch is up to date with 'progit/main'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   book/05-distributed-git/sections/contributing.asc
```

Teraz mając jeden plik gotowy do zacomittowania korzystamy z opcji -c ktora kopiuje metadane innego commita np.:

`git commit -c 1d32bd84467f316e628dbb186dca681b3250a0a3`

Powstały commit nie bedzie mojego autorstwa lecz oryginalnego autora a nawet z originalną datą
```
$ git show --pretty=fuller
commit 29d76cc0fdd380c6b1b7221241aeb8bbf4e5ccf8 (HEAD -> main)
Author:     Andrzej Wójtowicz <andre@amu.edu.pl>
AuthorDate: Fri Mar 4 14:47:25 2022 +0100
Commit:     Dawid Paluch <paluchdaw@gmail.com>
CommitDate: Mon Mar 20 08:59:04 2023 +0100

    Remove redundant new lines, fix formatting and typos

```

Aż tak skompilkowane to nie jest, a na ten moment nie wiem jak inaczej mozna by to ugryźć. Przykład pokazałem będąc na branchy main ale w rzeczywistości będzie to nasza własna, z której wystawiamy PR.

**Reply by olsza** [reply]
hmm, sprawdzę może, ale pewnie wyjdzie taki sam efekt, jak już zrobiłem w całości i jest dostępne na branch-u `test/main-pl` (wieczorem, po pracy, poprawię jedynie ścieżki do obrazków)
Ogólnie zrobiłem `git rebase main` na gałęzi master, a main w tym przypadku, jest oryginalny ze źródła ;) (ale też musiałem rezwiązać wszystkie konflikty... ale to też dło plus,że plików, których nie powinny być przeniesione, zostały przeniesione w odpowiednie miejsce zgodnie z oryginałem.
Ale tak wychodzi,ze Docker działa, autorzy z oryginalnego jak z naszego generują się, więc powinno być GIT ;)
